### ISSUE TITLE
Failing to parse a CSV file should print an error

### ISSUE THREAD

**Issue by nightblade9** [godotengine/godot#46682]
**Godot version:** 3.2.3

**OS/device including version:** Windows 10 PC

**Issue description:**
I added a translation CSV (hand-crafted in Notepad++) and didn't notice a comma in one of my strings. As a result, attempts to retrieve localized strings failed (most of them - the ones above the error line worked fine).

**Steps to reproduce:**
1) Create a CSV with an extraneous comma in it, such as:

```csv
key,en_US
FIRST,first line
HELLO,hello, world!
FAILED,not parsed
```

2) Add the localization resource in the project settings
3) Run the project. Note that there are no errors reported.

4) (Bonus) Try to access your localization strings:

- `tr("FIRST")` returns `first line`
- `tr("HELLO")` returns `HELLO`
- `tr("FAILED)` returns `FAILED`

It's very difficult to troubleshoot (you won't even know localization is broken if you look at strings before the error). To identify the problem, you have to do something like create a scene with lots of buttons (one per key).

It would be very useful if Godot would print out some kind of error indicating that there's an error with importing CSV files. (Best would be to print the name of the file with the error, and the line number.)

**Comment by Calinou** [comment]
Note that for more robust localization, I recommend looking at [using the gettext format](https://docs.godotengine.org/en/stable/tutorials/i18n/localization_using_gettext.html) instead.

**Comment by nightblade9** [comment]
I initially looked at gettext, but discarded that option for a few reasons:

- Requires additional tools to setup
- Harder to handle for someone new to localization

This is my first real Godot localization, simplicity and the ability to streamline the translation process is very important.

**Comment by progammer** [comment]
I'm referencing the docs right now, and it seems you can work around this [https://docs.godotengine.org/en/stable/getting_started/workflow/assets/importing_translations.html#doc-importing-translations](url) using quotes to wrap the strings that contain commas

**Comment by nightblade9** [comment]
> using quotes to wrap the strings that contain commas

Yes, this is exactly what I ended up doing. I don't think this is a problem, and I don't consider this a manual work-around - I think you *should* use quotes to guarantee consistency and ensure you don't make this mistake.

My issue is not quotes/no-quotes or use of a delimiter; my issue is that Godot didn't provide any error message suggesting to me where the problem might be. I spent a couple of hours retrying all the localization steps, creating a new project to reproduce the problem, and creating scenes with several messages in them, before I finally realized what was happening.

Those scenarios can apply regardless of your use of quotes or delimiter; Godot should print out an error if it encounters a problem in parsing. That might save someone else a couple hours of time tracing where exactly the problem is.
